File: 1-two-sum/solution.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer[]}
    def twoSum(self, nums, target):
        d = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in d:
                return d[diff]+1, i+1
            d[num] = i

File: 1-two-sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var d = {};
    for (var i=0; i < nums.length; ++i) {
        var diff = target - nums[i];
        if (d.hasOwnProperty(diff)) {
            return [d[diff]+1, i+1];
        }
        d[nums[i]] = i;
    }
};
